classes/Tools.php

```diff
@@ -611,22 +611,58 @@ public static function switchLanguage(Context $context = null)
             }
         }
     }
-
+    
+    /**
+     * Detects country and returns its ID
+     *
+     * @param null $address
+     *
+     * @return int
+     *
+     * @deprecated 1.7.4
+     */
     public static function getCountry($address = null)
     {
-        $id_country = (int)Tools::getValue('id_country');
-        if (!$id_country && isset($address) && isset($address->id_country) && $address->id_country) {
-            $id_country = (int)$address->id_country;
-        } elseif (Configuration::get('PS_DETECT_COUNTRY') && isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
-            preg_match('#(?<=-)\w\w|\w\w(?!-)#', $_SERVER['HTTP_ACCEPT_LANGUAGE'], $array);
-            if (is_array($array) && isset($array[0]) && Validate::isLanguageIsoCode($array[0])) {
-                $id_country = (int)Country::getByIso($array[0], true);
-            }
+        return self::getCountryId($address);
+    }
+
+    /**
+     * Detects country and returns its ID
+     * First tries GET parameter id_country, then id_country of the optional parameter $address,
+     * then detection from $_SERVER, then defaults to PS_COUNTRY_DEFAULT
+     *
+     * @param Address|null $address
+     *
+     * @return int
+     *
+     * @since 1.7.4
+     */
+    public static function getCountryId(Address $address = null)
+    {
+        if ($countryId = (int) Tools::getValue('id_country')) {
+            return $countryId;
+        }
+
+        if (null !== $address && $countryId = $address->id_country) {
+            return $countryId;
```

Comment:
I dislike the affectation in the test. What do you think about

```php
        if (null !== $address && $address->id_country) {
            return $address->id_country;
        }
```

?

Reply:
@Quetzacoalt91 Thank you! I made another commit.